File: examples/encrypt_token.py
import json
import base64
import os
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad

CIPHER = 'AES-256-CBC'
SECRET_KEY = b'MySuperSecretKeyForParamsToken12'  # 32 bytes key for AES-256

token_object = {
    "connection": {
        "type": "rdp",
        "settings": {
            "hostname": "10.0.0.12",
            "username": "Administrator",
            "password": "pAsSwOrD",
            "enable-drive": True,
            "create-drive-path": True,
            "security": "any",
            "ignore-cert": True,
            "enable-wallpaper": False
        }
    }
}


def encrypt_token(value):
    iv = os.urandom(16)  # 16 bytes for AES
    cipher = AES.new(SECRET_KEY, AES.MODE_CBC, iv)

    # Convert value to JSON and pad it
    padded_data = pad(json.dumps(value).encode(), AES.block_size)

    # Encrypt data
    encrypted_data = cipher.encrypt(padded_data)

    # Encode the IV and encrypted data
    data = {
        'iv': base64.b64encode(iv).decode('utf-8'),
        'value': base64.b64encode(encrypted_data).decode('utf-8')
    }

    # Convert the data dictionary to JSON and then encode it
    json_data = json.dumps(data)
    return base64.b64encode(json_data.encode()).decode('utf-8')


token = encrypt_token(token_object)

print("Parameters:")
print(json.dumps(token_object, indent=4))

print("\n\n")

print("Encrypted token:")
print(token)

print("\n\n")

print("Use this token in the URL:")
print("ws://localhost:8080/?token=" + token)

File: examples/encrypt_token.js
const crypto = require('crypto');

const CIPHER = 'aes-256-cbc';
const SECRET_KEY = 'MySuperSecretKeyForParamsToken12';

const tokenObject = {
    connection: {
        type: "rdp",
        settings: {
            "hostname": "10.0.0.12",
            "username": "Administrator",
            "password": "pAsSwOrD",
            "enable-drive": true,
            "create-drive-path": true,
            "security": "any",
            "ignore-cert": true,
            "enable-wallpaper": false
        }
    }
};

function encryptToken(value) {
    const iv = crypto.randomBytes(16);
    const cipher = crypto.createCipheriv(CIPHER, Buffer.from(SECRET_KEY), iv);

    let encrypted = cipher.update(JSON.stringify(value), 'utf8', 'base64');
    encrypted += cipher.final('base64');

    const data = {
        iv: iv.toString('base64'),
        value: encrypted
    };

    const json = JSON.stringify(data);
    return Buffer.from(json).toString('base64');
}

const originalToken = JSON.stringify(tokenObject, null, 2);
const token = encryptToken(tokenObject);
const urlencodedToken = encodeURIComponent(token);

console.log("Parameters:");
console.log(originalToken);

console.log("\n");

console.log("Encrypted token:");
console.log(token);

console.log("\n");

console.log("Use this token in the URL:");
console.log(`ws://localhost:8080/?token=${urlencodedToken}`);
